pypeit/fluxspec.py

```diff
@@ -444,22 +447,27 @@ def generate_sensfunc(self):
             return None
 
         ext_final = fits.getheader(self.par['std_file'], -1)
-        norder = ext_final['ECHORDER'] + 1
+        #norder = ext_final['ECHORDER'] + 1
+        norder = len(self.std)
 
         self.sens_dict = {}
         for iord in range(norder):
-            std_specobjs, std_header = load.load_specobjs(self.par['std_file'], order=iord)
-            std_idx = flux.find_standard(std_specobjs)
-            std = std_specobjs[std_idx]
+            #std_specobjs, std_header = load.load_specobjs(self.par['std_file'], order=iord)
+            #embed(header='447 of fluxspec')
+            #std_idx = flux.find_standard(std_specobjs)
+            std = self.std[iord] #std_specobjs[std_idx]
+
+            # THIS IS A CRAZY KLUDGE....
             try:
                 wavemask = std.optimal['WAVE_GRID'] > 0.0 #*units.AA
             except KeyError:
-                wavemask = std.optimal['WAVE'] > 1000.0 * units.AA
-                this_wave = std.optimal['WAVE'][wavemask]
+                wavemask = std.boxcar['WAVE'] > 1000.0 * units.AA
```

Comment:
I would prefer to not have quantities anywhere in PypeIt.

Reply:
am hesitant to remove this now.
you can do so as you refactor the fluxing.